io/eolearn/io/sentinelhub_service.py

```diff
@@ -355,12 +377,46 @@ def __init__(self, sen2cor_classification, layer, **kwargs):
             'SNW': FeatureType.DATA
         }
 
+        if 'image_format' in kwargs:
+            image_format = kwargs.get('image_format')
+            kwargs.pop('image_format', None)
+            sample_type = image_format.get_sample_type()
+        else:
+            image_format = MimeType.TIFF_d8
+            sample_type = 'INT8'
+
         if sen2cor_classification not in classification_types:
             raise ValueError('Unsupported Sen2Cor classification type: {}.'
                              ' Possible types are: {}'.format(sen2cor_classification, classification_types))
 
-        evalscript = 'return [{}];'.format(sen2cor_classification)
+        evalscript = '''\
```

Comment:
I suggest moving this evalscript template outside the class and have it as a constant (maybe even in a separate module?). Defining it here in the middle of the method looks weird.

Reply:
General idea (with upcoming Processing API) is that evalscripts will be extracted in the way you describe here. As we are not there yet, I believe the evalscript pertaining to extracting SCL masks is better defined here. I suggest creating a new module for evalscripts as a new PR.

Reply:
Ok, then we can leave it as it is for now.